vault/diagnose/tls_verification.go

```diff
@@ -21,17 +20,32 @@ const maxVersionError = "'tls_max_version' value %q not supported, please specif
 
 // ListenerChecks diagnoses warnings and the first encountered error for the listener
 // configuration stanzas.
-func ListenerChecks(ctx context.Context, listeners []listenerutil.Listener) ([]string, []error) {
+func ListenerChecks(ctx context.Context, listeners []*configutil.Listener) ([]string, []error) {
+	testName := "check-listener-tls"
+	ctx, span := StartSpan(ctx, testName)
+	defer span.End()
 
 	// These aggregated warnings and errors are returned purely for testing purposes.
 	// The errors and warnings will report in this function itself.
 	var listenerWarnings []string
 	var listenerErrors []error
 
-	for _, listener := range listeners {
-		l := listener.Config
+	for _, l := range listeners {
 		listenerID := l.Address
 
+		if l.TLSDisable {
+			Warn(ctx, fmt.Sprintf("listener at address: %s has error %s. ", listenerID, "TLS is disabled in a Listener config stanza"))
```

Comment:
did you want these trailing spaces here (for like concat future strings on or something), or is it a typo?

Reply:
also, does it make sense to just const the string that's being added (or just include it in the first string if not)?

Reply:
Nope! I'll remove the trailing spaces! Yeah, good point -- I'll just include it in the first string.

Reply:
cool! mark as resolved whenever you do and i'll re-review